lang/rust/Cargo.toml

```diff
@@ -34,7 +34,7 @@ authors = ["Apache Avro team <dev@avro.apache.org>"]
 license = "Apache-2.0"
 repository = "https://github.com/apache/avro"
 edition = "2021"
-rust-version = "1.70.0"
+rust-version = "1.73.0"
```

Comment:
According to https://github.com/serde-rs/bytes/blob/0.11.15/Cargo.toml#L12 the MSRV for serde_bytes is 1.53.
Why do we need to bump to 1.73.0 here ?

Reply:
This is required by `std::thread_local!`'s `LocalKey` when wrapped in a `Cell` for having get/set methods available, as you can see [in the doc](https://doc.rust-lang.org/std/thread/struct.LocalKey.html#method.get). Bumping MSRV was actually advised by clippy.